java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowFileWriter.java

```diff
@@ -76,15 +74,8 @@ public ArrowFileWriter(VectorSchemaRoot root, DictionaryProvider provider, Writa
   }
 
   public ArrowFileWriter(VectorSchemaRoot root, DictionaryProvider provider, WritableByteChannel out,
-                         Map<String, String> metaData, IpcOption option, CompressionCodec.Factory compressionFactory,
-                         CompressionUtil.CodecType codecType) {
-    this(root, provider, out, metaData, option, compressionFactory, codecType, Optional.empty());
-  }
-
-  public ArrowFileWriter(VectorSchemaRoot root, DictionaryProvider provider, WritableByteChannel out,
-                         Map<String, String> metaData, IpcOption option, CompressionCodec.Factory compressionFactory,
-                         CompressionUtil.CodecType codecType, Optional<Integer> compressionLevel) {
-    super(root, provider, out, option, compressionFactory, codecType, compressionLevel);
+      Map<String, String> metaData, IpcOption option, CompressionCodec codec) {
```

Comment:
Isn't this removing a public constructor?

Reply:
You can delegate constructors without removing public ones (which breaks API).

Reply:
@lidavidm does the updated change make sense?